inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/support/AnnotationQueueOverviewDataProvider.java

```diff
@@ -300,50 +321,100 @@ public long getInProgressAmountForDocument(
 
     //Returns a random document out of all documents in the project.
     //Only a document is chosen which is not yet given to annotators more than the default number
-    //per document number
-    public SourceDocument getRandomDocument()
+    //per document number (TODO),
+    //However, if there was already a document in progress,
+    // this will always be returned (can only be one)
+
+    public SourceDocument getRandomDocument(
+        AnnotationPageBase aAnnotationPageBase, AnnotationDocument aCurrentAnno)
     {
-        //Create an empty document
-        SourceDocument document = null;
-        Random r = new Random();
 
-        while (document == null)
-        {
-            int i = r.nextInt(data.size() - 1);
-            //If the random chosen document wont surpass the amount of default number combining
-            // "inProgress" for the document + "finished" amount for the document + 1
-            if ((getInProgressAmountForDocument(data.
-                get(i)) +
-                getFinishedAmountForDocument(data.
-                    get(i)))
-                + 1 <= defaultAnnotations)
-            {
-                //If that was not the case, assign this document
-                document = data.get(r.nextInt(data.size() - 1));
+
+        //First check for a documents that is in Progress for the user, and return this
+        for (int i = 0; i < allAnnotationDocuments.size(); i++) {
+            //NULL CHECK due to listAnnotableDocuments from webanno
+            if (allAnnotationDocuments.get(i) != null &&
+                allAnnotationDocuments.get(i).getState().equals(IN_PROGRESS)
+                && !allAnnotationDocuments.get(i).getName().equals(aCurrentAnno.getName())) {
+                //Found a document in progress
+                for (SourceDocument doc: data) {
+                    if (doc.getName().equals(allAnnotationDocuments.get(i).getName())) {
+                        return doc;
+                    }
+                }
+            }
+        }
+
+        //Nothing in Progress, so now return a random document, or create a new annotation document
+
+
+        //Get a random document
+        List<SourceDocument> srcList = entityManager.
+            createQuery(
+                "FROM SourceDocument " +
+                "WHERE project = :project " +
+                "ORDER BY rand()", SourceDocument.class)
+            .setParameter("project", aAnnotationPageBase.getModelObject().getProject())
```

Comment:
get the AnnotationState once at the beginning of the method and then use it instead of repeatedly calling annotationpagebase.getModelObject()